app/controllers/posts_controller.rb

```diff
@@ -658,9 +699,27 @@ def unless_locked
   end
 
   def do_draft_delete(path)
-    key = "saved_post.#{current_user.id}.#{path}"
+    body_key = "saved_post.#{current_user.id}.#{path}"
+    comment_key = "saved_post.#{current_user.id}.#{path}.comment"
+    excerpt_key = "saved_post.#{current_user.id}.#{path}.excerpt"
+    license_key = "saved_post.#{current_user.id}.#{path}.license"
+    tags_key = "saved_post.#{current_user.id}.#{path}.tags"
+    tag_name_key = "saved_post.#{current_user.id}.#{path}.tag_name"
+    title_key = "saved_post.#{current_user.id}.#{path}.title"
     saved_at = "saved_post_at.#{current_user.id}.#{path}"
-    RequestContext.redis.del key, saved_at
+
+    keys = [
+      body_key,
+      comment_key,
+      excerpt_key,
+      license_key,
+      tags_key,
+      tag_name_key,
+      title_key,
+      saved_at
+    ]
```

Comment:
Also feels like this could be a map over just the part names - at least for the specific parts, with the body key added after.

```ruby
[:comment, :excerpt, :license].map { |key| "saved_post.#{current_user.id}.#{path}.#{key.to_s}" }
```

(only some fields included, others left as an exercise for the reader :) )

Reply:
Good idea - I was adding ghem one-by-one, so never noticed they ended up matching 1:1, will do!